src/app/dashboard/[interval]/[[...date]]/components/NavigationButton.tsx

```diff
@@ -0,0 +1,49 @@
+import Link from "next/link";
+import { Button } from "@/components/ui/button";
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import { cn } from "@/lib/utils";
+
+interface NavigationButtonProps {
+  href: string;
+  direction: "prev" | "next";
+  isVisible: boolean;
+}
+
+export function NavigationButton({
+  href,
+  direction,
+  isVisible,
+}: NavigationButtonProps) {
+  const Icon = direction === "prev" ? ChevronLeft : ChevronRight;
+  const label = direction === "prev" ? "Previous" : "Next";
+
+  return (
+    <>
+      {/* Desktop */}
+      <Button
+        variant="outline"
+        asChild
+        className={cn("hidden items-center sm:flex", !isVisible && "invisible")}
+      >
+        <Link href={href}>
+          {direction === "prev" && <Icon className="mr-2 h-4 w-4" />}
+          <span>{label}</span>
+          {direction === "next" && <Icon className="ml-2 h-4 w-4" />}
+        </Link>
+      </Button>
+
+      {/* Mobile */}
```

Comment:
This is a pretty generic component not specific to the dashboard pages. Should be defined in the src/components folder instead and reused in the other places where we have this same type of component. 

The only components / files that should be in the `src/app/<page>/**` directories are components that are specific to that page / business logic.